tower-http/src/content_encoding.rs

```diff
@@ -60,30 +61,34 @@ impl Encoding {
         }
     }
 
-    // based on https://github.com/http-rs/accept-encoding
     pub(crate) fn from_headers(
         headers: &HeaderMap,
         supported_encoding: impl SupportedEncodings,
     ) -> Self {
-        let mut preferred_encoding = None;
-        let mut max_qval = 0.0;
-
-        for (encoding, qval) in encodings(headers, supported_encoding) {
-            if (qval - 1.0f32).abs() < 0.01 {
-                preferred_encoding = Some(encoding);
-                break;
-            } else if qval > max_qval {
-                preferred_encoding = Some(encoding);
-                max_qval = qval;
-            }
-        }
-
-        preferred_encoding.unwrap_or(Encoding::Identity)
+        preferred_encoding(encodings(headers, supported_encoding).iter())
+            .unwrap_or(Encoding::Identity)
     }
 }
+/// Select the max quota Encoding from a list of (Encoding,quota) or None when list is empty.
+/// When any one of all quota equals, the last one selected.
+/// It will select br for Chrome as Chrome usually has a header value "Accept-Encoding: gzip, deflate, br".
```

Comment:
[PR 175](https://github.com/tower-rs/tower-http/pull/177) will pick first Encoding, that is gzip, when Header "Accept-Encoding: gzip, deflate, br" is processed. it will fallback to uncompressed if path/file.ext.gz not exists but path/file.ext.br DOES exist. It is expected to fallback to br.

Reply:
Brotli is introduced by Google, it should be served first, then gzip, the third deflate, in my opinion.
The new algorithm select the last one (not the first one) when all quota equals. You know "Accept-Encoding: gzip, deflate, br" will be parsed as a list of [(gzip,1),(deflate,1),(br,1)], the quota of each Encoding is 1.

Reply:
I understand the purpose of the PR but not why its necessary to serve br first specifically for chrome.